shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-env/src/test/java/org/apache/shardingsphere/test/integration/env/container/atomic/storage/DockerStorageContainer.java

```diff
@@ -78,6 +78,26 @@ protected void configure() {
                         : DataSourceEnvironment.getURL(databaseType, "localhost", getFirstMappedPort()), getUsername(), getUnifiedPassword())));
     }
     
+    protected void setCommands(final String[] commands) {
+        if (commands.length > 0) {
+            for (String each : commands) {
+                setCommand(each);
+            }
+        }
+    }
+    
+    protected void addEnvs(final Map<String, String> envs) {
+        if (!envs.isEmpty()) {
+            envs.forEach(this::addEnv);
+        }
+    }
+    
+    protected void mapResources(final Map<String, String> resources) {
```

Comment:
Please add final if method does not design for extension

Reply:
done